Source/WebCore/page/Quirks.cpp

```diff
@@ -2335,6 +2335,33 @@ std::optional<Quirks::TikTokOverflowingContentQuirkType> Quirks::needsTikTokOver
     return { };
 }
 
+// rdar://166400170
+bool Quirks::needsInstagramResizingReelsQuirk(const Element& element, const RenderStyle& elementStyle, const RenderStyle& parentStyle) const
+{
+    if (!needsQuirks()) [[unlikely]]
+        return { };
+
+    if (!m_quirksData.quirkIsEnabled(QuirksData::SiteSpecificQuirk::NeedsInstagramResizingReelsQuirk))
+        return { };
+
+    if (elementStyle.display() != DisplayType::Block)
+        return { };
+
+    if (elementStyle.isOverflowVisible())
+        return { };
+
+    if (!elementStyle.width().isAuto())
+        return { };
+
+    if (parentStyle.display() != DisplayType::Flex)
+        return { };
```

Comment:
You could probably also group this with the display block, and then the width checks together.

Reply:
I wanted to keep the checks for the element itself and then the parent together